### PR TITLE
feat: Allow JSONata based transformations in JSON serializer

### PR DESCRIPTION
**Pull request by srebhan**
- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

replaces #11205

This PR adds a new `json_transformation` option to the JSON serializer. This new option can be used to transform the JSON output using the [JSONata language](https://jsonata.org/) to accommodate for requirements on the receiver side. Additional benefits are the various filtering and processing capabilities of the introduced language (see README for a few basic examples).

### CODE REVIEW

**Comment by srebhan** [comment]
@reimda so you _approve_ it? :-D

File: go.mod [hunk 1]
```diff
@@ -166,6 +166,8 @@ require (
 	modernc.org/sqlite v1.10.8
 )
 
+require github.com/blues/jsonata-go v1.5.4
```
Comment (sspaink) [inline go.mod:169]: Can you put this in the above require block?

File: plugins/serializers/json/json_test.go [hunk 2]
```diff
@@ -23,7 +29,7 @@ func TestSerializeMetricFloat(t *testing.T) {
 	}
 	m := metric.New("cpu", tags, fields, now)
 
-	s, _ := NewSerializer(0, "")
+	s, _ := NewSerializer(0, "", "")
```
Comment (sspaink) [inline plugins/serializers/json/json_test.go:32]: The returned error is ignored throughout the test, think we should check it?

**Review by sspaink — CHANGES_REQUESTED** [review]
This looks really useful! two minor comments, but I think it looks good.

**Review by sspaink — APPROVED** [review]

**Review by reimda — APPROVED** [review]

**Pull request merged**